src/components/Radio/Radio.module.css

```diff
@@ -180,7 +180,7 @@
     border-color: var(--eds-theme-color-border-utility-critical);
   }
 
-  & ~ .radio__selected {
+  &:checked ~ .radio__selected {
```

Comment:
Fixes bug where selected treatment is applied to ALL radio buttons regardless of checked state